lib/tasks/licenses.rake

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+namespace :licenses do
+
+  report_path = Rails.root.join('doc/licenses/licenses.xml')
+
+  desc 'Generates a report with the dependencies and their licenses'
+  task :report do
+    warn 'Generating report...'
+    system license_finder_report(report_path), exception: true
+  end
+
+  desc 'Check license compliance of dependencies'
+  task :compliance do
+    warn 'Checking license compliance...'
+    unless system('license_finder action_items')
+      warn '*** License compliance test failed  ***'
+      exit 1
+    end
+
+    warn 'Checking license report is up to date...'
+    temp_report_path = Tempfile.new(report_path.to_s).path
+    system license_finder_report(temp_report_path)
+
+    unless identical?(report_path, temp_report_path)
+      warn '*** License report outdated. Please run "rails licenses:report" ***'
+      exit 1
+    end
+  end
+
+  private
+
+  def license_finder_report(path)
+    "license_finder report --format=xml --save=#{path}"
```

Comment:
btw we don't have to run an external process to generate the report. We can do

```ruby
require 'license_finder'
lf = LicenseFinder::CLI::Main.new
lf.options = { "format" => "html", "save" => "/tmp/api.html" }
lf.report
```

I think it is preferable to avoid calling external programs. But if you strongly prefer keep calling it like this, I will not insist :innocent: